src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts

```diff
@@ -45,19 +122,37 @@ export const load = async ({ parent, params }) => {
       }
     );
     if (matchedServiceGroupsFromSlug && matchedServiceGroupsFromSlug.length > 0) {
+      let matchedPageMetadata;
       // account for possibility that two service groups have the same ending slug
-      const matchedServiceGroupsMetadata = matchedServiceGroupsFromSlug.map((group) => {
-        const serviceGroupMetadataId = group?.pageMetadata?.sys?.id;
-        if (serviceGroupMetadataId) return pageMetadataMap.get(serviceGroupMetadataId);
+      const matchedServiceGroup = matchedServiceGroupsFromSlug.find((group) => {
+        const serviceGroupMetadataId = group?.pageMetadata?.sys?.id || "";
+        matchedPageMetadata = pageMetadataMap.get(serviceGroupMetadataId);
+        if (matchedPageMetadata) {
+          return matchedPageMetadata.url === url;
+        }
       });
-      // Regardless of how many matches we have, we want to ensure the URL matches.
-      //   Otherwise we could render the page for /land/fire/safety at /foo/bar/safety.
-      // TODO: Possibly account for possiblity that two service groups (erroneously) have the same URL
-      const matchedPageMetadata = matchedServiceGroupsMetadata.find(
-        (matchedServiceGroupMetadata) => matchedServiceGroupMetadata?.url === url
-      );
-      if (matchedPageMetadata) {
-        return { pageMetadata: matchedPageMetadata };
+
+      if (matchedServiceGroup) {
+        const serviceEntries = matchedServiceGroup.serviceEntriesCollection?.items.filter(
+          (item) => item?.__typename === "ServiceEntry"
+        );
+        let serviceGroups = matchedServiceGroup.serviceEntriesCollection?.items.filter(
+          (item) => item?.__typename === "ServiceGroup"
+        ) as ServiceGroup[];
```

Comment:
This is a small nit so it's not worth blocking the PR for, but for future reference it's safer to use a type generated from the query or explicitly defined where you need it rather than casting to a type from the schema, since the schema includes _all_ fields, even those that aren't included in the query.

Reply:
ah, that makes sense. I was trying to avoid casting, but resolving the TS errors without it was proving to be really difficult. I'm down to try to improve that aspect as follow-up